modules/neuwoRtdProvider.js

```diff
@@ -0,0 +1,168 @@
+import { deepAccess, deepSetValue, generateUUID, logError, logInfo, mergeDeep } from '../src/utils.js';
+import { getRefererInfo } from '../src/refererDetection.js';
+import { ajax } from '../src/ajax.js';
+import { submodule } from '../src/hook.js';
+import * as events from '../src/events.js';
+import CONSTANTS from '../src/constants.json';
+
+export const DATA_PROVIDER = 'neuwo.ai';
+const SEGTAX_IAB = 2 // IAB - Content Taxonomy version 2
+const RESPONSE_IAB_TIER_1 = 'marketing_categories.iab_tier_1'
+const RESPONSE_IAB_TIER_2 = 'marketing_categories.iab_tier_2'
+
+function init(config = {}, userConsent = '') {
+  config.params = config.params || {}
+  // ignore module if publicToken is missing (module setup failure)
+  if (!config.params.publicToken) {
+    logError('publicToken missing', 'NeuwoRTDModule', 'config.params.publicToken')
+    return false;
+  }
+  return true;
+}
+
+export function getBidRequestData(reqBidsConfigObj, callback, config, userConsent) {
+  config.params = config.params || {};
+  logInfo('NeuwoRTDModule', 'starting getBidRequestData')
+
+  const wrappedArgUrl = encodeURIComponent(config.params.argUrl || getRefererInfo().page);
+  const url = 'https://m1apidev.neuwo.ai/edge/GetAiTopics?' + [
+    'token=' + config.params.publicToken,
+    'lang=en',
+    'url=' + wrappedArgUrl
+  ].join('&')
+  let billingId = generateUUID();
+  events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, { type: 'auction', billingId, auctionId: reqBidsConfigObj.auctionId, vendor: neuwoRtdModule.name })
```

Comment:
how is the billing event useful? Can't you just bill on api call activity?

Reply:
Events are required for integration with Magnite.